elide-core/src/main/java/com/yahoo/elide/core/pagination/Pagination.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core.pagination;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+
+import javax.ws.rs.core.MultivaluedMap;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Encapsulates the pagination strategy.
+ */
+
+@AllArgsConstructor
+@ToString
+public class Pagination {
+
+    public static final int DEFAULT_PAGE_SIZE = 500;
+    // todo do we ever want to return more than 10000 items? Set Upper Bounds
+
+    private static final Pagination DEFAULT_PAGINATION = new Pagination(0, 0);
+
+    public static final Map<String, String> PAGE_KEYS = new HashMap<>();
+    static {
+        PAGE_KEYS.put("page[size]", "pageSize");
```

Comment:
@aklish The idea was to support everything other than cursor for datastore's with concrete indicies. So whether you add just page[number]=2, then it would start at 500-1000 if using default page size (not sure if the default page size should be 500 or if that should move to the 10,000 we discussed), if you add page[offset]=50,page[limit]=50, then you would get 50-100 from the records, etc.